qutip/solver/__init__.py

```diff
@@ -0,0 +1,4 @@
+from .evolver import *
```

Comment:
Should we not go for absolute imports and explicitly writing down all imports if not too many? This is the convention for most other imports in QuTiP. 
```
from qutip.solver.evolver import ... 
```

Reply:
In dev.major we swapped to preferring relative imports everywhere in qutip when you're staying within the same package, so this is the consistent choice.  Absolute imports disguise circular import issues, because it's much less obvious when you're leaving a package, and particularly with us rearranging a lot of internal modules, absolute imports make it much harder to modify this sort of thing. The absolute imports remaining should only be where legacy code is circular, in cython `cimport`, or when accessing a different package. In the latter case, you shouldn't be specifying the internal structure though, you should only be importing from the public API, because everything else is liable to change.

It's also a lot of typing for a questionably negative outcome, especially if it ends up in user code - a user should never have to know what file we defined `tensor` in, they should just be able to do `qutip.tensor`. There's no benefit to making them type `from qutip.core.tensor import tensor`, and we make no guarantees that we won't rearrange the internal structure at some point, so if they do type that, it may break in a later version. It's the same thing between separate packages.

Reply:
What about the use of the `*`, should we not be specific about which names we are importing?